# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/FFTBlock.cpp
# Commit: 9b4ea6ecadd1
# Full Hash: 9b4ea6ecadd1e56b9a279feab220a8bfab36809c
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-28 16:14:38
# Regressor Bug: 1879873
# File Overlap Count: 1
# Description:
#   Bug 1879873 - Directly use tx instead of the older avfft interface. r=karlt,media-playback-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202156
# ==============================================================================

diff -r 7c71c3d1a738 -r 9b4ea6ecadd1 dom/media/webaudio/FFTBlock.cpp
--- a/dom/media/webaudio/FFTBlock.cpp	Wed Feb 28 12:50:27 2024 +0000
+++ b/dom/media/webaudio/FFTBlock.cpp	Wed Feb 28 12:50:27 2024 +0000
@@ -30,13 +30,14 @@
 
 #include "FFTBlock.h"
 
+#include "FFVPXRuntimeLinker.h"
 #include <complex>
 
 namespace mozilla {
 
-using Complex = std::complex<double>;
+FFmpegFFTFuncs FFTBlock::sFFTFuncs = {};
 
-FFmpegRDFTFuncs FFTBlock::sRDFTFuncs = {nullptr, nullptr, nullptr};
+using Complex = std::complex<double>;
 
 static double fdlibm_cabs(const Complex& z) {
   return fdlibm_hypot(real(z), imag(z));
@@ -58,7 +59,8 @@
   int fftSize = newBlock->FFTSize();
   AlignedTArray<float> buffer(fftSize);
   newBlock->GetInverseWithoutScaling(buffer.Elements());
-  AudioBufferInPlaceScale(buffer.Elements(), 1.0f / fftSize, fftSize / 2);
+  AudioBufferInPlaceScale(buffer.Elements(),
+                          1.0f / AssertedCast<float>(fftSize), fftSize / 2);
   PodZero(buffer.Elements() + fftSize / 2, fftSize / 2);
 
   // Put back into frequency domain.